macheteLogger/src/main/java/com/virtuslab/logger/MacheteLogger.java

```diff
@@ -0,0 +1,34 @@
+package com.virtuslab.logger;
+
+import java.util.function.Supplier;
+
+import kr.pe.kwonnam.slf4jlambda.LambdaLogger;
+
+public class MacheteLogger implements IMacheteLogger {
+  private final LambdaLogger LOG;
+  private final String className;
+
+  MacheteLogger(LambdaLogger logger) {
+    LOG = logger;
+    // We are sure that at the moment when this constructor is called we have at least 4 elements in stacktrace array
+    @SuppressWarnings("upperbound")
+    StackTraceElement element = Thread.currentThread().getStackTrace()[3];
+    String[] classPath = element.getClassName().split("\\.");
+    className = classPath[classPath.length - 1];
+  }
+
+  private String getMethodReferenceName() {
+    // We are sure that at the moment when this method is called we have at least 7 elements in stacktrace array
+    @SuppressWarnings("upperbound")
+    StackTraceElement element = Thread.currentThread().getStackTrace()[6];
```

Comment:
Similar here, drop a comment on what's expected in elements 0...5
That's a pretty brittle construct TBH, so we should at least keep track of what's expected in the latest stack frames while we're in this method